File: retrocast/internal/models/guild.go
package models

import "time"

type Guild struct {
	ID        int64     `json:"id,string"`
	Name      string    `json:"name"`
	IconHash  *string   `json:"icon_hash,omitempty"`
	OwnerID   int64     `json:"owner_id,string"`
	CreatedAt time.Time `json:"created_at"`
}

File: retrocast-client/Retrocast/Sources/Models/Guild.swift
import Foundation

struct Guild: Codable, Identifiable, Hashable, Sendable {
    let id: Snowflake
    var name: String
    var iconHash: String?
    let ownerID: Snowflake
    let createdAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case iconHash = "icon_hash"
        case ownerID = "owner_id"
        case createdAt = "created_at"
    }
}
